src/main/java/hello/controller/BookingUIController.java

```diff
@@ -0,0 +1,141 @@
+package hello.controller;
+
+import hello.exceptions.BadRequestException;
+import hello.exceptions.NotFoundException;
+import hello.model.Apartment2;
+import hello.model.Booking;
+import hello.model.BookingRequest;
+import hello.model.Customer;
+import hello.service.ApartmentService;
+import hello.service.BookingService;
+import hello.service.CustomerService;
+import org.apache.log4j.Logger;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Controller;
+import org.springframework.ui.Model;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.ModelAttribute;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+
+
+import java.awt.print.Book;
+import java.util.List;
+import java.util.NoSuchElementException;
+
+@Controller
+public class BookingUIController {
+
+    private BookingService bookingService;
+    private ApartmentService apartmentService;
+    private CustomerService customerService;
+
+
+    @Autowired
+    public BookingUIController(BookingService bookingService, ApartmentService apartmentService, CustomerService customerService) {
+        this.bookingService = bookingService;
+        this.apartmentService = apartmentService;
+        this.customerService = customerService;
+
+
+    }
+
+    static Logger log = Logger.getLogger(CustomerController.class);
+
+
+    @GetMapping("/booking")
+    public String showBookingLandingPage(Model model) {
+        model.addAttribute("activeLink", "Booking");
+        return "booking";
+    }
+
+
+    @GetMapping("/newBookingCreate")
+    public String showAddBookingForm(Model model) {
+        BookingRequest booking = new BookingRequest();
+        model.addAttribute("apartments", apartmentService.getAllApartments());
+        model.addAttribute("customers", customerService.getAllCustomers());
+        model.addAttribute("activeLink", "Booking");
+        model.addAttribute("booking", booking);
+
+        return "newBookingCreate";
+    }
+
+    @PostMapping(value = "/newBooking")
+    public String showBooking(@ModelAttribute("booking") BookingRequest booking, Model model) {
+        try {
+            List<Customer> c = customerService.findByName(booking.name);
+            if (c.isEmpty()) {
+                log.info("Customer name does not exist");
+                return "badRequest";
+            }
+
+            Booking newBooking = new Booking(c.get(0), booking.apartment);
```

Comment:
What happens if there are multiple customers with the same name? Here, we're making an assumption and just taking the first customer